includes/Shopping/Woocommerce_Query.php

```diff
@@ -76,9 +75,17 @@ public function get_search( string $search_term ) {
 		_prime_post_caches( $products_image_ids, false, true );
 
 		foreach ( $products as $product ) {
-			$product_image_ids = array_map( 'absint', $product->get_gallery_image_ids() );
-			$images            = array_map( [ $this, 'get_product_images' ], $product_image_ids );
+			$product_image_ids = array_merge( [ $product->get_image_id() ], $product->get_gallery_image_ids() );
+			$product_image_ids = array_map( 'absint', $product_image_ids );
+			$product_image_ids = array_unique( array_filter( $product_image_ids ) );
+
+			$images = array_map(
+				[ $this, 'get_product_image' ],
+				$product_image_ids
+			);
 
+			$images = array_filter( $images );
```

Comment:
should not need this line now, this is done above on line 80.

Reply:
Can we work out why we have getting null values here. That doesn't seem right

Reply:
I can dig in more later but likely because I'm returning `[]` if there's no url from the map

Reply:
Outdated --- the updated code doesn't seem to need it.